mbedtls/src/pk/mod.rs

```diff
@@ -295,16 +246,16 @@ impl Pk {
         Ok(ret)
     }
 
-    pub fn private_from_ec_components(mut curve: EcGroup, private_key: Mpi) -> Result<Pk> {
+    pub fn private_from_ec_components<F: Random>(rng: &mut F, mut curve: EcGroup, private_key: Mpi) -> Result<Pk> {
```

Comment:
This one is odd. We need to understand why is it algorithmically necessary to have an RNG here. Presumably, in order to blind operations for constant-time when performing the scalar mult.

Reply:
You are correct.
There was a to-do comment in the `EcPoint::mul` for adding rng for `mul`: https://github.com/fortanix/rust-mbedtls/blob/f9175bdfa634685d3f2457cc417f8ed3dab79f6a/mbedtls/src/ecp/mod.rs#L327
And in mbedtls 3.X, the rng callback no longer could be null: https://github.com/Mbed-TLS/mbedtls/blob/v3.2.1/include/mbedtls/ecp.h#L931
So I add rng here.

Reply:
Added comments to explain why it need a RNG now